src/table/cells/handle-cell-focus.js

```diff
@@ -89,6 +89,57 @@ export const handleNavigateTop = ({ tableSectionRef, focusedCellCoord, rootEleme
   }
 };
 
+export const getCellSelectionStatusNote = (rows, translator) =>
+  rows.length === 1
+    ? translator.get('SNTable.SelectionLabel.OneSelectedValue')
+    : translator.get('SNTable.SelectionLabel.SelectedValues', [rows.length]);
+
+export const getMemoisedSrNotation = (prevCount = 0) => {
+  let prevSelectedCount = prevCount || 0;
+  let hasJunkChar = 0;
+
+  return ({ focusedCellCoord, rootElement, selState, translator, isActiveElementInTable }) => {
+    if (!focusedCellCoord || focusedCellCoord[0] === 0 || !isActiveElementInTable) return '';
+
+    const [rowIdx, colIdx] = focusedCellCoord;
+    const rowElements = rootElement.getElementsByClassName('sn-table-row');
+    const cell = rowElements[rowIdx]?.getElementsByClassName('sn-table-cell')[colIdx];
+
+    const isCellSelected = cell && cell.classList.contains('selected');
+
+    let notation = '';
+
+    if (selState.rows.length) {
+      const selectionNote = getCellSelectionStatusNote(selState.rows, translator);
+
+      if (prevSelectedCount < selState.rows.length) {
+        // if we select cell
+        notation += `${translator.get('SNTable.SelectionLabel.SelectedValue')} ${selectionNote}`;
+      } else if (prevSelectedCount > selState.rows.length) {
+        // if we deselect cell
+        notation += `${translator.get('SNTable.SelectionLabel.DeselectedValue')} ${selectionNote}`;
+      } else if (isCellSelected) {
+        // if we are in selection mode and move to selected cell
+        notation += translator.get('SNTable.SelectionLabel.SelectedValue');
+      } else {
+        // if we are in selection mode and move to unselected cell
+        notation += translator.get('SNTable.SelectionLabel.NotSelectedValue');
+      }
+
+      // Junk char addition
+      if (hasJunkChar % 2) notation += ` ­`;
+      hasJunkChar++;
+    } else if (selState.rows.length === 0 && prevSelectedCount > 0) {
+      // if we deselect last (selected) cell which means we close the selection mode
+      notation += translator.get('SNTable.SelectionLabel.ExitedSelectionMode');
+    }
+
+    prevSelectedCount = selState.rows.length;
+    return notation;
+  };
+};
+
+export const getCellSrNotation = getMemoisedSrNotation();
```

Comment:
This logic gets confusing in this way. Would a hook not be a better solution? Then you can keep the previous state as well. Probably good to add a refactor task for this as well

Reply:
maybe, i'm not sure if we could bring functionality with hook, but of course i'll create a refactor issue for that. 👌

Reply:
added this one on the github board: https://github.com/qlik-oss/sn-table/issues/332